### PR TITLE
feat: Test coverage automation 

### LINKED ISSUE #132: Test coverage automation
**Is your feature request related to a problem? Please describe.**
There is no automated process that gives a coverage report of the tests when submitting a PR.
If there was one it would be easier to maintain the quality and would allow  being  in control of the actual coverage changes in real time when submitting new pieces of code.

**Describe the solution you'd like**
1.Clean up tests.
2. Integration with Codecov.
3. add coverage badge to "readme"

**Describe alternatives you've considered**
It could be done manually but in is complicated and not justified.  

**Additional context**

### PR DESCRIPTION
**Pull request by GabrielSamberg**
Closes #132 

## 📑 Description
Linked the Codecov automated service to the relevant GitHub action workflow. now by submitting a PR there will be also an automatic  coverage report via Codecov in addition to the automated tests.

here is a link to an example of how this might look (this has been done on my own repository):
https://app.codecov.io/github/GabrielSamberg/keep/commit/ab7203267dd15d3351972b0e6216d7cfb07ca552/indirect-changes

In order for this to work on the main repository here is the link to the app installation:
https://github.com/apps/codecov 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: .github/workflows/test-pr.yml [hunk 1]
```diff
@@ -11,9 +11,16 @@ jobs:
     - name: Install Poetry
       uses: snok/install-poetry@v1
     - run: poetry install
-    - run: poetry run coverage run -m pytest
+    - run: poetry run coverage run --branch -m pytest
+    - run: poetry run coverage json
     - name: Archive code coverage results
       uses: actions/upload-artifact@v3
       with:
         name: code-coverage-report
```
Comment (TheCoreMan) [inline .github/workflows/test-pr.yml:19]: Delete the entire step, not just the file path. There's no need to upload any artifact.

File: .github/workflows/test-pr.yml [hunk 2]
```diff
@@ -11,9 +11,16 @@ jobs:
     - name: Install Poetry
       uses: snok/install-poetry@v1
     - run: poetry install
-    - run: poetry run coverage run -m pytest
+    - run: poetry run coverage run --branch -m pytest
+    - run: poetry run coverage json
```
Comment (TheCoreMan) [inline .github/workflows/test-pr.yml:15]: I think adding names to these steps will make the action itself look better, like so:

```suggestion
    - name: Run unit tests and report coverage
      run: poetry run coverage run --branch -m pytest
    - name: Convert coverage results to JSON (for CodeCov support)
      run: poetry run coverage json
```

**Review by TheCoreMan — CHANGES_REQUESTED** [review]
See the two comments + we should rebase the commits to re-word so they fit into conventional commits + squash the two updates to the test-pr file.

**Comment by talboren** [comment]
@TheCoreMan @GabrielSamberg lets also add a coverage badge to readme, wdyt?

**Review by TheCoreMan — APPROVED** [review]
LGTM :)

**Review by talboren — APPROVED** [review]
LGTM! 🚀

**Pull request merged**